src/components/Icon/Icon.js

```diff
@@ -11,28 +11,47 @@ import { keen } from "style/theme";
  */
 export default function Icon({
   ariaLabelledby,
-  hasBadge,
   badgeColor,
   badgePosition,
-  borderColor,
-  borderWidth,
+  borderColor, // Deprecate
+  borderWidth, // Deprecate
   children,
-  fillColor,
+  fill,
+  fillColor, // Deprecate
   fillOpacity,
   glyphName,
+  hasBadge,
   size,
+  stroke,
+  strokeOpacity,
+  strokeWidth,
   ...props
 }) {
+
+  if (borderWidth !== undefined) {
+    console.warn("Use `strokeWidth` rather than `borderWidth`. `borderWidth` will be deprecated in the next major version.");
+  }
+  if (borderColor !== undefined) {
+    console.warn("Use `stroke` rather than `borderColor`. `borderColor` will be deprecated in the next major version.");
+  }
+  if (fillColor !== undefined) {
+    console.warn("Use `fill` rather than `fillColor`. `fillColor` will be deprecated in the next major version.");
+  }
+
```

Comment:
Here is a cleaner way to do this. To make this work you'll also need to remove the deprecated fields from the list of prop arguments (lines 16-17, 20).

```suggestion
const deprecatedFields = {borderWidth: "strokeWidth", borderColor: "stroke", fillColor: "fill"}

Object.keys(deprecatedFields).forEach(df => props[df] !== undefined && console.warn(`Use ${deprecatedFields[df]} rather than ${df}. ${df} will be deprecated in the next major version.`))
```